synadm/api.py

```diff
@@ -1139,3 +1140,47 @@ def user_shadow_ban(self, user_id, unban):
         else:
             method = "post"
         return self.query(method, f"v1/users/{user_id}/shadow_ban")
+
+    def notice_send(self, receivers, content_plain, content_html, paginate, to_regex):
+        """ Send server notices.
+
+        Args:
+            receivers: Target(s) of the notice. Either localpart or regular
+                expression matching localparts.
+            content_plain: Unformatted text of the notice.
+            content_html: HTML-formatted text of the notice.
+        """
+        data = {
+            "user_id": "",
+            "content": {
+                "msgtype": "m.text",
+                "body": content_plain,
+                "format": "org.matrix.custom.html",
+                "formatted_body": content_html
+            }
+        }
+
+        # A regular expression was supplied to match receivers.
+        if to_regex:
+            outputs = []
+            response = self.user_list(0, paginate, True, False, "", "")
```

Comment:
Also I was thinking: We are doing this only when regex option given, why not always check if the user is existing before showing the confirmation prompt? Performance reasons?

Reply:
I don't think there was a specific reason for me to not do it for single recipients. I would suggest to change the `else` branch to the following:
```
else:
            data["user_id"] = generate_mxid(receivers)
            return [self.query("post", f"v1/send_server_notice", data=data)]
```
In my opinion checking whether the user exists is not necessary in our code as that is really something synapse can do best and gives good error messages for. But the "input checking" part (illegal characters, generating mxid from localpart) can and should be done by synadm - and it's achieved by this minor change.

Reply:
Can't work like that. generate_mxid() is not available in SynapseAdmin object. You are alreading passing through it in notice.py. Need to revert this commit: https://github.com/JOJ0/synadm/pull/74/commits/3f8b78e5cbd40213d3f60c0a7d64032e389276ae